packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -3566,29 +3570,20 @@ class _ExitWidgetSelectionButtonGroupState extends State<_ExitWidgetSelectionBut
     if (buttonBuilder == null) {
       return null;
     }
+    final ValueNotifier<bool> selectionOnTapEnabledNotifier =
+        WidgetsBinding.instance.debugWidgetInspectorSelectionOnTapEnabledNotifier;
 
-    return _TooltipGestureDetector(
+    return _WidgetInspectorButton(
       button: buttonBuilder(
         context,
         onPressed: () {
-          final bool defaultTapBehaviorEnabled =
-              WidgetsBinding.instance.debugWidgetInspectorDefaultTapBehaviorEnabled;
-          WidgetsBinding.instance.debugWidgetInspectorDefaultTapBehaviorEnabled =
-              !defaultTapBehaviorEnabled;
-          WidgetInspectorService.instance.selection.clear();
+          _changeSelectionOnTapMode();
         },
         semanticLabel: 'Change widget selection mode for taps',
-        defaultTapBehaviorEnabled:
-            WidgetsBinding.instance.debugWidgetInspectorDefaultTapBehaviorEnabled,
+        defaultTapBehaviorEnabled: selectionOnTapEnabledNotifier.value,
       ),
       onTooltipVisible: () {
```

Comment:
just pass this as a tearoff: `onTooltipVisible: _changeSelectionOnTapTooltip`

Reply:
oops thanks!